applications/DG-Max/Algorithms/DivDGMaxEigenvalue.cpp

```diff
@@ -185,15 +187,27 @@ class DivDGMaxEigenvalue<DIM>::SolverWorkspace {
         err = EPSCreate(PETSC_COMM_WORLD, &solver_);
         CHKERRABORT(PETSC_COMM_WORLD, err);
         // Setting operators, but these will be set again after k-shifting
-        // Note the order Mx = (1/omega^2) Sx and NOT Sx = omega^2 M x and doing
-        // a spectral transformation.
-        // It may be possible to remove this relic with the current version of
-        // SLEPc.
-        err = EPSSetOperators(solver_, massMatrix_, stiffnessMatrix_);
+        err = EPSSetOperators(solver_, stiffnessMatrix_, massMatrix_);
+        CHKERRABORT(PETSC_COMM_WORLD, err);
+
+        // The mass matrix is positive semidefinite, the stiffness matrix is
+        // Hermitian indefinite (k-phase-shifts are also Hermitian for real k).
+        err = EPSSetProblemType(solver_, EPS_GHEP);
```

Comment:
do we have to bump the Slepc version in cmake?

Reply:
No, all the needed functionality is long there in SLEPc. But probably nobody knew about it or something like that.